source/iface/nd_item.rst

```diff
@@ -12,174 +10,259 @@
+
+
+::
+
+  size_t get_global_range(int dimension) const
 
-Returns the :ref:`range` of the index space.
+Return the same value as ``get_global_range().get(dimension).``.
 
 ``get_local_range``
 ===================
 
 ::
 
-  sycl::range<dimensions> get_local_range() const;
-  size_t get_local_range(int dimension) const;
+  range<Dimensions> get_local_range() const
+
+Returns a ``range`` representing the dimensions
+of the current work-group.
+
+
+::
+
+  size_t get_local_range(int dimension) const
 
-Returns the position of the work item in the work group.
+Return the same value as ``get_local_range().get(dimension)``.
 
 ``get_offset``
 ==============
 
 ::
 
-  sycl::id<dimensions> get_offset() const;
+  id<Dimensions> get_offset() const
+    // Deprecated in SYCL 2020.
 
-Returns the offset provided to the :ref:`handler-parallel_for`.
+Deprecated in SYCL 2020. Returns an id representing the
+n-dimensional offset provided to the constructor of the
+``nd_range`` and that is added by the runtime to
+the global id of each work-item.
 
 ``get_nd_range``
 ================
 
 ::
 
-  sycl::nd_range<dimensions> get_nd_range() const;
+  nd_range<Dimensions> get_nd_range() const
 
-Returns the :ref:`nd_range` provided to the :ref:`handler-parallel_for`.
+Returns the ``nd_range`` of the current execution.
 
-``barrier``
-===========
+``async_work_group_copy``
+=========================
 
 ::
 
-  void barrier(sycl::access::fence_space accessSpace =
-    sycl::access::fence_space::global_and_local) const;
-
-Executes a work group barrier.
+  template <typename DataT>
+  device_event async_work_group_copy(local_ptr<DataT> dest,
+                                     global_ptr<DataT> src,
+                                     size_t numElements) const
 
+  template <typename DataT>
+  device_event async_work_group_copy(global_ptr<DataT> dest,
+                                     local_ptr<DataT> src,
+                                     size_t numElements) const
 
-``mem_fence``
-=============
+  template <typename DataT>
+  device_event async_work_group_copy(local_ptr<DataT> dest,
+                                     global_ptr<DataT> src,
+                                     size_t numElements, size_t srcStride) const
 
-::
+  template <typename DataT>
+  device_event async_work_group_copy(global_ptr<DataT> dest,
+                                     local_ptr<DataT> src,
+                                     size_t numElements, size_t destStride) const
 
-  template <sycl::access::mode accessMode = sycl::access::mode::read_write>
-  void mem_fence(sycl::access::fence_space accessSpace =
-    sycl::access::fence_space::global_and_local) const;
+All four ``async_work_group_copy`` functions listed above are deprecated
+in SYCL 2020. Has the same effect as the overload taking
+``decorated_local_ptr`` and ``decorated_global_ptr``
+except that the ``dest`` and ``src`` parameters are
+``multi_ptr#s with [code]#access::decorated::legacy``.
```

Comment:
Formatting is broken here

Reply:
Could you outline more details? I see in output it looks fine but didn't catch wrong part...

Reply:
![image](https://github.com/KhronosGroup/SYCL_Reference/assets/34314961/609492ba-60b5-496a-8818-95c3174e984a)

In my render, there are '#' and '[code]' in the output, and the word "with" is in code font when I don't think it should be.  Are you seeing differently in your render?

Reply:
@mkinsner could you recheck is spec correct?
[https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#table.members.nditem](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#table.members.nditem)

Deprecated in SYCL 2020. Has the same effect as the overload taking decorated_local_ptr and decorated_global_ptr except that the dest and src parameters are multi_ptr#s with [code]#access::decorated::legacy.


Please put here corrected version, I was thinking that spec is fine...
The same problem for group class in [
Table 118. Member functions for the group class
](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#table.members.group)

Reply:
Thanks for pointing this out.  I've opened a PR to fix the spec:
https://github.com/KhronosGroup/SYCL-Docs/pull/518

Reply:
Removed 's' with : https://github.com/KhronosGroup/SYCL_Reference/pull/122/commits/6bdee50f6363fd9a9e1fb925605895e377ce0b13

Reply:
Done with https://github.com/KhronosGroup/SYCL_Reference/pull/122/commits/fcc33dcf4832c0c5cbf1d220d17bafb05b972616